tests/std/tests/P0408R7_efficient_access_to_stringbuf_buffer/test.cpp

```diff
@@ -0,0 +1,245 @@
+        Stream stream{move(s)};
+        s = move(stream).str();
+        stream.str(move(s));
+        assert(*pcount == 1);
+    }
+};
+
+template <template <typename> typename Test>
+void run_counting_test() {
+    run_test_util<Test<counting_stringstream>>();
+    run_test_util<Test<counting_istringstream>>();
+    run_test_util<Test<counting_ostringstream>>();
+}
+
+template <const char* str>
+void test_iterator_dereference_death() {
+    string s      = str;
+    auto iterator = s.begin();
+    stringstream stream{move(s)};
+    (void) *iterator; // cannot dereference invalid iterator
+}
+
+template <const char* str>
+void test_iterator_operator_arrow_death() {
+    string s      = str;
+    auto iterator = s.begin();
+    stringstream stream{move(s)};
+    (void) iterator.operator->(); // cannot dereference invalid iterator
+}
+
+template <const char* str>
+void test_iterator_increment_death() {
+    string s      = str;
+    auto iterator = s.begin();
+    stringstream stream{move(s)};
+    ++iterator; // cannot increase invalid iterator
+}
+
+void test_iterator_increment_zero(const char* str) {
+    string s      = str;
+    auto iterator = s.begin();
+    stringstream stream{move(s)};
+    iterator += 0; // OK
+}
+
+class test_stringbuf : public stringbuf {
+public:
+    using stringbuf::basic_stringbuf;
+
+    using stringbuf::eback;
+    using stringbuf::egptr;
+    using stringbuf::epptr;
+    using stringbuf::gptr;
+    using stringbuf::pbase;
+    using stringbuf::pptr;
+};
+
+void test_init_buf_ptrs_out(test_stringbuf& buf) {
+    assert(*buf.pbase() == large_string[0]);
+    assert(buf.epptr() >= buf.pbase() + buf.view().size());
+    assert(buf.pptr() == buf.pbase());
+}
+
+void test_init_buf_ptrs_out_ate(test_stringbuf& buf) {
+    assert(*buf.pbase() == large_string[0]);
+    assert(buf.epptr() >= buf.pbase() + buf.view().size());
+    assert(buf.pptr() == buf.pbase() + buf.view().size());
+}
+
+void test_init_buf_ptrs_in(test_stringbuf& buf) {
+    assert(*buf.eback() == large_string[0]);
+    assert(buf.gptr() == buf.eback() && buf.egptr() == buf.eback() + buf.view().size());
+}
+
+void test_init_buf_ptrs() {
+    test_stringbuf buf1{string{large_string}, ios_base::out};
+    test_init_buf_ptrs_out(buf1);
+
+    test_stringbuf buf2{string{large_string}, ios_base::out | ios_base::ate};
+    test_init_buf_ptrs_out_ate(buf2);
+
+    test_stringbuf buf3{string{large_string}, ios_base::in};
+    test_init_buf_ptrs_in(buf3);
+
+    test_stringbuf buf4{string{large_string}, ios_base::in | ios_base::out};
+    test_init_buf_ptrs_out(buf4);
+    test_init_buf_ptrs_in(buf4);
+
+    test_stringbuf buf5{string{large_string}, ios_base::in | ios_base::out | ios_base::ate};
+    test_init_buf_ptrs_out_ate(buf5);
+    test_init_buf_ptrs_in(buf5);
+}
+
+int main(int argc, char* argv[]) {
+    std_testing::death_test_executive exec([] {
+        run_test<test_rvalue>();
+        run_test<test_allocator>();
+
+#if _ITERATOR_DEBUG_LEVEL == 0

```

Comment:
this can be a regular if statement I think.

Reply:
It follows the pattern of other tests.

https://github.com/microsoft/STL/blob/530bdc5aaa8a21277e1281ad3df8b8d8433b5caa/tests/std/tests/VSO_0000000_string_view_idl/test.cpp#L214

Reply:
allright, in that case I don't care that much.